File: C/pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    // asks for change owed and validates input, only accepts positive numbers
    float dollars;
    do
    {
        dollars = get_float("Change owed in dollars: ");
    }
    while (dollars <= 0);
    
    int coins = 0;
    // converts from float to int for precision
    int cents = round(dollars * 100);
    
    // loop continues until you are owed no change
    while (cents > 0)
    {
        // conditionals check for every type of coin, subtract that coin from the total change and add 1 to the total number of coins
        if (cents >= 25)
        {
            cents -= 25;
            coins++;
        }
        else if (cents >= 10)
        {
            cents -= 10;
            coins++;
        }
        else if (cents >= 5)
        {
            cents -= 5;
            coins++;
        }
        else
        {
            cents -= 1;
            coins++;
        }
    }
    // displays the total number of coins needed to give the change
    printf("you are owed %i coins\n", coins);
}

File: Python/pset6/cash/cash.py
from cs50 import get_float

# Asks for user input until it receives a valid one
dollars = 0
while (dollars <= 0):
    dollars = get_float("Change owed: ")

coins = 0
cents = round(dollars * 100, 0)

# count coins until the change owed is 0
while (cents > 0):
    if (cents >= 25):
        cents -= 25
        coins += 1
    elif (cents >= 10):
        cents -= 10
        coins += 1
    elif (cents >= 5):
        cents -= 5
        coins += 1
    else:
        cents -= 1
        coins += 1

# displays the total number of coins needed to give the change
print(f"you are owed {coins} coins")